lib/mumukit/metatest/test_result_builder.rb

```diff
@@ -0,0 +1,20 @@
+module Mumukit::Metatest
+  class TestResultBuilder
+    attr_accessor :title, :status, :result, :summary_type, :summary_message
+
+    def summary
+      {type: summary_type.presence, message: summary_message.presence}.compact
+    end
+
+    def build
+      raise 'missing status' unless status
+      raise "invallid #{status}" unless status.passed? || status.failed?
+
+      if summary_message.present? || summary_type.present?
```

Comment:
maybe `unless summary.empty?`

Reply:
Hmm, I'd rather not do that because that means creating two useless hashes in most situations - the literal hash plus the compact hash.

Reply:
Also, I think is less intention revealing. I wan't here to state that the builder will create only a 4-tuple - with a summary - when there is at least one of the summary fields it's designed to collect.

Reply:
:+1: